test/specs/repo/special-property.ts

```diff
@@ -55,18 +55,20 @@ describe( 'Special:NewProperty', function () {
 		);
 		await SpecialNewProperty.submit();
 
-		// wait for the $wgWBRepoSettings['sharedCacheDuration'] cache to
-		// timeout, so the list of properties reflects the change
-		// eslint-disable-next-line wdio/no-pause
-		await browser.pause( 2000 );
-
-		await SpecialListProperties.openParams( {
-			dataType: wikibasePropertyString.urlName,
-			limit: 1000
+		let numberOfPropertiesAfter;
+		await browser.waitUntil( async () => {
+			await SpecialListProperties.openParams( {
+				dataType: wikibasePropertyString.urlName,
+				limit: 1000
+			} );
+			numberOfPropertiesAfter = await SpecialListProperties.properties.length;
+			return numberOfPropertiesAfter === numberOfPropertiesBefore + 1;
+		}, {
+			// wait for the $wgWBRepoSettings['sharedCacheDuration'] cache to
+			// timeout, so the list of properties reflects the change
+			interval: 1000,
+			timeout: 10000,
+			timeoutMsg: 'expected total number of reflect new property in list within 10 seconds'
```

Comment:
^^ Similar to `mocha.retries()` this is utilising the WDIO `waitUntil()` method to do a similar thing but within the test. Note that the page will reload at the top of the loop. The default timeout for `waitUntil` is 5 seconds, and the interval set here is 1 second. So this will try to pass once Immediately, and again at 1, 2, 3, 4, and 5 seconds, before failing.